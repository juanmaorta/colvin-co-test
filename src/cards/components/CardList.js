import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Grid from '@material-ui/core/Grid'

import BlackCard from './BlackCard'

class CardList extends Component {
  editCard = (cardId) => {
    console.log(`editing card ${cardId}`)
  }

  deleteCard = (cardId) => {
    console.log(`deleting card ${cardId}`)
  }

  render () {
    const {
      cards
    } = this.props

    return (
      <Grid container spacing={40}>
        {cards.map((card, idx) => {
          return (
            <BlackCard
              card={card}
              key={idx}
              editable
              handleDelete={this.deleteCard}
              handleEdit={this.editCard}
            />
          )
        })}
      </Grid>
    )
  }
}

CardList.propTypes = {
  cards: PropTypes.array.isRequired
}

export default CardList
