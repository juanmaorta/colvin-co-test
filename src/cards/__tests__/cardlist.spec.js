/* global expect, describe, it */
import React from 'react'
import { shallow } from 'enzyme'

import CardList from '../components/CardList'
import * as stubs from './stubs'

describe('CardList snapshots', () => {
  it('should match for CardList component', () => {
    const wrapper = shallow(
      <CardList
        cards={stubs.cards}
        deleteCard={jest.fn}
        editCard={jest.fn}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
