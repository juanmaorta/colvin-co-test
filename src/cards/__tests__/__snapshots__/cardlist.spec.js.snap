// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList snapshots should match for CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardList
    cards={
        Array [
            Object {
              "id": 1,
              "text": "Goldfinger: He's the man, the man with...",
            },
            Object {
              "id": 2,
              "text": "Instead of coal, Santa now gives the bad children...",
            },
            Object {
              "id": 3,
              "text": "Knock knock. Who's there?",
            },
          ]
    }
    classes={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(BlackCard)
          card={
                    Object {
                              "id": 1,
                              "text": "Goldfinger: He's the man, the man with...",
                            }
          }
          editable={true}
          handleDelete={[Function]}
          handleEdit={[Function]}
/>,
        <WithStyles(BlackCard)
          card={
                    Object {
                              "id": 2,
                              "text": "Instead of coal, Santa now gives the bad children...",
                            }
          }
          editable={true}
          handleDelete={[Function]}
          handleEdit={[Function]}
/>,
        <WithStyles(BlackCard)
          card={
                    Object {
                              "id": 3,
                              "text": "Knock knock. Who's there?",
                            }
          }
          editable={true}
          handleDelete={[Function]}
          handleEdit={[Function]}
/>,
      ],
      "container": true,
      "spacing": 40,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "id": 1,
            "text": "Goldfinger: He's the man, the man with...",
          },
          "editable": true,
          "handleDelete": [Function],
          "handleEdit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "id": 2,
            "text": "Instead of coal, Santa now gives the bad children...",
          },
          "editable": true,
          "handleDelete": [Function],
          "handleEdit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "card": Object {
            "id": 3,
            "text": "Knock knock. Who's there?",
          },
          "editable": true,
          "handleDelete": [Function],
          "handleEdit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(BlackCard)
            card={
                        Object {
                                    "id": 1,
                                    "text": "Goldfinger: He's the man, the man with...",
                                  }
            }
            editable={true}
            handleDelete={[Function]}
            handleEdit={[Function]}
/>,
          <WithStyles(BlackCard)
            card={
                        Object {
                                    "id": 2,
                                    "text": "Instead of coal, Santa now gives the bad children...",
                                  }
            }
            editable={true}
            handleDelete={[Function]}
            handleEdit={[Function]}
/>,
          <WithStyles(BlackCard)
            card={
                        Object {
                                    "id": 3,
                                    "text": "Knock knock. Who's there?",
                                  }
            }
            editable={true}
            handleDelete={[Function]}
            handleEdit={[Function]}
/>,
        ],
        "container": true,
        "spacing": 40,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "id": 1,
              "text": "Goldfinger: He's the man, the man with...",
            },
            "editable": true,
            "handleDelete": [Function],
            "handleEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "id": 2,
              "text": "Instead of coal, Santa now gives the bad children...",
            },
            "editable": true,
            "handleDelete": [Function],
            "handleEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "card": Object {
              "id": 3,
              "text": "Knock knock. Who's there?",
            },
            "editable": true,
            "handleDelete": [Function],
            "handleEdit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
